pkg/apis/autoscaling/v1/conversion.go

```diff
@@ -383,6 +391,15 @@ func Convert_v1_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler(i
 		delete(out.Annotations, autoscaling.MetricSpecsAnnotation)
 	}
 
+	if behaviorEnc, hasConstraints := out.Annotations[autoscaling.BehaviorSpecsAnnotation]; hasConstraints {
```

Comment:
out.Annotations[autoscaling.BehaviorSpecsAnnotation] may have been assigned on line 355.
In that case, is json.Unmarshal call below necessary ?

Reply:
@tedyu: Probably I didn't get your question, could you rephrase it?

line 355 is part of `Convert_autoscaling_HorizontalPodAutoscaler_To_v1_HorizontalPodAutoscaler` function. I.e., it converts the `autoscaling.HorizontalPodAutoscaler` object of the current version (with behavior) to the v1 version. The v1 version doesn't have behaviors (it's a base version), so that's why we put `behavior` into annotations.

The current line (394), is part of `Convert_v1_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler`, i.e. it converts the object back from v1 to the current version. And if there's information about behavior in annotation, we convert it into a usual `autoscaling.HorizontalPodAutoscalerBehavior` field (check line 399).

Reply:
Due to the way diff is showed, I didn't pay attention to the two func's (as you described).

Thanks for explaining.